lua/jmod/sv_util.lua

```diff
@@ -570,14 +570,18 @@ function JMod.ShouldAllowControl(self, ply)
 	return (engine.ActiveGamemode() ~= "sandbox" or ply:Team() ~= TEAM_UNASSIGNED) and ply:Team() == self.Owner:Team()
 end
 
+
 function JMod.ShouldAttack(self, ent, vehiclesOnly, peaceWasNeverAnOption)
+   	local Gaymode, PlayerToCheck, InVehicle=engine.ActiveGamemode(), nil, false
+    local snpcs = ent.CPTBase_NPC == true or ent.IsDrGNextbot or ent.IsVJBaseSNPC == true
+    local lfs = IsValid(self.Owner:Team()) and ent.LFS and ent:GetAITEAM() == self.Owner:Team()
```

Comment:
don't use IsValid on a potential number, just do a truthy check
also, put self.Owner:Team() in a var beforehand instead of calling the getter twice
also, instead of checking for LFS, just check for .GetAITEAM